lib/Reporters/MutationTestingElementsReporter.cpp

```diff
@@ -36,30 +36,17 @@ json11::Json createFiles(const Result &result,
   // on their source file.
   // Step 1: we create a map: "file => all of its mutation points".
   std::map<std::string, std::vector<MutationPoint *>> mutationPointsPerFile;
-  for (auto &mutationResult : result.getMutationResults()) {
-    MutationPoint *mutant = mutationResult->getMutationPoint();
-
+  for (auto mutant : result.getMutationPoints()) {
```

Comment:
Here is the actual bug: a mutation point may have more than one mutation result. If we execute 5 different tests against one mutant, then we get 5 mutation results for this one mutation point.
Therefore, if we collect all mutation points for each mutation result, then we end up with 5 mutants, which are in fact 5 copies of the same mutant.

Correct behavior here is to go through mutation points since they are unique.

Reply:
That was my point from the very beginning. It is good that now we are on the same page even though you point about the template-based code producing different results still applies.

I have reused this code from the IDEReporter without looking too much into how it should be done:

```
  for (auto &mutationResult : result.getMutationResults()) {
    auto mutant = mutationResult->getMutationPoint();
    auto &executionResult = mutationResult->getExecutionResult();
```